### DISCUSSION TITLE
How to get the current Project directory path

### DISCUSSION THREAD

**Discussion by DefaultLP** [tauri-apps/tauri#7308] [Q&A] [answered]
I'm currently working on an app that deals with file systems a lot. I wanted to set up an exemplary one in my project folder for my own convenience. It is found on the same level as the `package.json`. So it is included in my git repository.

Now I wanted to ask if there is an intelligent way to get said project directory path. The best thing I found was to use the **RESOURCE** dir and work my way up from there.

There is one discussion #7199 that mentions this. There the solution was to use Rusts cwd. From my testing (printing it in the console) this leads to `project_dir/src-tauri`. This is sort of in the same vein. Except I have to traverse up only one instead of thrice.

Also on a related note. Is it better to do file searching (like filter out all images in a directory) in **Rust** or **js/ts**? 
I'm currently doing all the file related searches in frontend since it is conceptually easier for me to reason about.

**Comment by FabianLars** [comment, marked as answer]
> There is one discussion https://github.com/tauri-apps/tauri/discussions/7199 that mentions this. There the solution was to use Rusts cwd. From my testing (printing it in the console) this leads to project_dir/src-tauri. This is sort of in the same vein. Except I have to traverse up only one instead of thrice.

Yeah, so using rust is really the closest you'll get to it since that's the actual cwd of the program. While tauri's cli is typically aware of the project's root the tauri crate is _not_, Can't think of an _automatic_ way to get the dir you're expecting rn 🤔 

> Also on a related note. Is it better to do file searching (like filter out all images in a directory) in Rust or js/ts?
I'm currently doing all the file related searches in frontend since it is conceptually easier for me to reason about.

With v1's IPC performance i'd do it in Rust (though to be fair i do everything in rust...) but if you're not throttled by speed and find it easier i'd say there's not much wrong about doing it in js (except the possibility to leak the whole fs structure to malicious js code, afaik only really a concern if your app interacts with something on the internet)

**Reply by DefaultLP** [reply]
Thanks a lot for these answers! I will switch my code to Rust in the future. The main reason I haven't done so is mainly my nearly decade-long expertise in Javascript, compared to my not even half a year of experience in Rust. But since I wanted to learn more about Rust regardless I take this as an opportunity.

Thanks again!
